SharpPcap/LibPcap/CaptureFileWriterDevice.cs

```diff
@@ -199,12 +199,7 @@ public void Write(RawCapture p)
             Write(data, ref header);
         }
 
-        void IInjectionDevice.SendPacket(RawCapture p)
-        {
-            Write(p);
-        }
-
-        void IInjectionDevice.SendPacket(ReadOnlySpan<byte> p)
+        void IInjectionDevice.SendPacket(ReadOnlySpan<byte> p, ICaptureHeader header)
```

Comment:
Signature mismatch between interface and implementation
```suggestion
        void IInjectionDevice.SendPacket(ReadOnlySpan<byte> p, ICaptureHeader header = null)
```

Reply:
not seeing a compile time warning here for that, is it defining a different method due to the default parameter difference?

Reply:
Tried to add = null and seeing, warning CS1066: The default value specified for parameter 'header' will have no effect because it applies to a member that is used in contexts that do not allow optional arguments

Reply:
I am not 100% sure here what magic the compiler is doing, but I would rather avoidany side effects from parameter mismatch

Reply:
@kayoub5 alright, its because we specify that the method is part of the interface:

void IInjectionDevice.SendPacket(ReadOnlySpan<byte> p, ICaptureHeader header)

because we are saying we are implementing the IInjectionDevice SendPacket() method, and IInjectionDevice says that header is optional its not able to be used in any other context. I'd probably prefer we not add the warning by trying to over specify it.

Oh and it looks like c# 8.0 has some mechanism we could use to solve this via:

https://stackoverflow.com/a/23141985/299037